shapely/geometry/collection.py

```diff
@@ -35,11 +37,18 @@ def __init__(self, geoms=None):
           >>> l = LineString([(52, -1), (49, 2)])
           >>> gc = GeometryCollection([p, l])
         """
-        BaseMultipartGeometry.__init__(self)
         if not geoms:
-            pass
-        else:
-            self._geom, self._ndim = geos_geometrycollection_from_py(geoms)
+            # TODO better empty constructor
+            return pygeos.from_wkt("GEOMETRYCOLLECTION EMPTY")
```

Comment:
For all those constructors, I now use WKT to create an empty geometry, but see https://github.com/pygeos/pygeos/issues/149 to discuss if we need to have a better way for this